x-pack/test/functional/apps/ml/anomaly_detection/annotations.ts

```diff
@@ -7,96 +7,336 @@
+        await ml.jobAnnotations.assertAnnotationContentById(annotationId, expectedEditedAnnotation);
+      });
+
+      it('displays newly edited annotation in anomaly explorer and job list', async () => {
+        await ml.testExecution.logTestStep('should display edited annotation in anomaly explorer');
+        await ml.navigation.navigateToAnomalyExplorerViaSingleMetricViewer();
+        await ml.anomalyExplorer.assertAnnotationsPanelExists('loaded');
+
+        await ml.jobAnnotations.ensureAnomalyExplorerAnnotationsPanelOpen();
+        await ml.jobAnnotations.assertAnnotationContentById(annotationId, expectedEditedAnnotation);
+
+        await ml.testExecution.logTestStep('should display edited annotation in job list');
+        await ml.navigation.navigateToJobManagement();
+        await ml.jobTable.waitForJobsToLoad();
+        await ml.jobTable.filterWithSearchString(JOB_CONFIG.job_id, 1);
+        await ml.jobTable.openAnnotationsTab(JOB_CONFIG.job_id);
+        await ml.jobAnnotations.assertAnnotationContentById(annotationId, expectedEditedAnnotation);
+      });
+    });
+
+    describe('deleting', function () {
+      const annotationId = `delete-annotation-id-${Date.now()}`;
+
+      before(async () => {
+        await ml.api.indexAnnotation(annotation, annotationId);
+      });
+
+      it('displays the original annotation', async () => {
+        await ml.testExecution.logTestStep('loads from job list row link');
+        await ml.navigation.navigateToMl();
+        await ml.navigation.navigateToJobManagement();
+
+        await ml.jobTable.waitForJobsToLoad();
+        await ml.jobTable.filterWithSearchString(JOB_CONFIG.job_id, 1);
+
+        await ml.jobTable.clickOpenJobInSingleMetricViewerButton(JOB_CONFIG.job_id);
+        await ml.commonUI.waitForMlLoadingIndicatorToDisappear();
+
+        await ml.testExecution.logTestStep('pre-fills the job selection');
+        await ml.jobSelection.assertJobSelection([JOB_CONFIG.job_id]);
+
+        await ml.testExecution.logTestStep('pre-fills the detector input');
+        await ml.singleMetricViewer.assertDetectorInputExist();
+        await ml.singleMetricViewer.assertDetectorInputValue('0');
+        await ml.singleMetricViewer.assertAnnotationsExists('loaded');
+
+        await ml.testExecution.logTestStep('displays annotation content');
+        await ml.jobAnnotations.ensureSingleMetricViewerAnnotationsPanelOpen();
+        await ml.jobAnnotations.assertAnnotationsRowExists(annotationId);
+        await ml.jobAnnotations.assertAnnotationContentById(annotationId, {
+          annotation: annotation.annotation,
+          from: '2016-02-10 22:13:51',
+          to: '2016-02-11 14:24:49',
+          modifiedTime: '2021-07-12 22:38:18',
+          modifiedBy: annotation.modified_username,
+          event: annotation.event,
+        });
+      });
+
+      it('deletes the annotation', async () => {
+        await ml.testExecution.logTestStep('opens edit annotation flyout');
```

Comment:
I think the test step description is not accurate for this test. But given that the whole test only contains one high level action and one assertion, I think we could remove the test step completely from this test (in tests with multiple actions the test step logging helps to easily find where things went wrong in case of a test failure).

Reply:
Updated here https://github.com/elastic/kibana/pull/105527/commits/ec07c2c1e48391ce4acbdba4ddc5d8d8fa4593f5